tests/agent_server/deploy/test_scheduler_agent.py

```diff
@@ -170,6 +173,39 @@ async def pass_method():
     pass
 
 
+class TestScheduler(ResourceScheduler):
+    def __init__(self, environment: uuid.UUID, executor_manager: executor.ExecutorManager[executor.Executor], client: Client):
+        super().__init__(environment, executor_manager, client)
+
+    async def should_be_running(self, endpoint: str) -> bool:
+        return True
+
+    def _create_agent(self, agent: str, should_start: bool = True) -> None:
+        """Start processing for the given agent"""
+        self._workers[agent] = TestTaskRunner(endpoint=agent, scheduler=self)
+        self._workers[agent].start()
```

Comment:
with some changes, we may be able to get awayt with just replacing the should_run method